MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -193,6 +193,14 @@ open class RouteLegProgress: NSObject {
     
     @objc public var userHasArrivedAtWaypoint = false
     
+    /**
+     An approximatation of the closest segment on the current leg.
+     */
+    @objc public var currentSegment: Int {
```

Comment:
What does this value mean? A segment is a kind of geometry, not a number.

Reply:
It describes the user's current index into coordinates or annotations on a leg.

Reply:
The comment and property name are both misleading, then. It’s the current segment _index_.

Reply:
This is very different than how we handle per-segment congestion attributes. I think we should unify the approach: either distribute the speed limit attributes among each step, or change the congestion logic to depend on this `currentSegmentIndex` property and stop distributing the attributes among the steps.